from sqlite3 import connect, Error
from lib.io.network import Network


class DBHelper(object):
    def __init__(self):
        self.__cursor = None
        self.__connexion = None

    def close(self):
        if self.__connexion:
            self.__connexion.close()

    def connect(self):
        # Credits for database https://github.com/j91321/rext
        self.__connexion = connect('./resources/db/oui.db')
        self.__cursor = self.__connexion.cursor()

    def fetchone(self, IP):
        self.connect()
        if Network.ping(IP):
            mac, oui = Network.arp(IP)
            return mac, self.query(oui)
        else:
            return None, None

    def query(self, oui):
        data = (oui,)
        self.__cursor.execute('SELECT * FROM oui WHERE oui LIKE ?', data)
        try:
            data = self.__cursor.fetchone()
        except Error:
            data = None

        return data
